dynamiqs/time_array.py

```diff
@@ -604,6 +638,14 @@ def dtype(self) -> jnp.dtype:
     def shape(self) -> tuple[int, ...]:
         return self.f.shape
 
+    @property
+    def dims(self) -> tuple[int, ...]:
+        return self.f(0.0).dims
+
+    @property
+    def ndiags(self) -> int:
+        return self.f(0.0).ndiags
+
```

Comment:
A `jax.eval_shape(f).dims` or `jax.eval_shape(f).ndiags` should do the trick without evaluating the function here.